app/database_etl/airtable_records_handler/airtable_records_formatter.py

```diff
@@ -105,3 +120,21 @@ def apply_study_max_estimate_grade(df: pd.DataFrame) -> pd.DataFrame:
                 subset['estimate_grade'] = level
                 continue
     return df
+
+
+def batch_update_airtable_records(records_to_update, field_names):
+    AIRTABLE_API_KEY = os.getenv('AIRTABLE_API_KEY')
+    AIRTABLE_BASE_ID = os.getenv('AIRTABLE_BASE_ID')
+
+    # Create table object
+    table = Table(AIRTABLE_API_KEY, AIRTABLE_BASE_ID, 'Rapid Review: Estimates')
+
+    # Cycle through all the records to update
```

Comment:
`iterrows` broadly recommended against for perf reasons. What I might suggest is turning the inside of the loop into a function, and then using `df.apply`. See relevant answer here: https://stackoverflow.com/questions/24870953/does-pandas-iterrows-have-performance-issues

Reply:
I believe df.apply works if we are applying a function to a df and we want the output to be within the existing df. That's why the function returns the df with the function applied. I think that's what vectorization typically allows you to do - apply a transformation to a matrix as an entire operation instead of going row by row. In this case, we don't care about generating an output and storing it in the original records_to_update df. We want to cycle through the df and create a dictionary from it. Not sure that df.apply makes sense in this case for that reason. I do know there are performance issues, which is why vectorization is preferred when you are applying functions to the df across all rows and cols.

Reply:
`df.apply` returns a `pd.series` object where the values of the series are the outputs of the function specified, applied to each row of the input df. It doesn't necessarily need to involve storing the result in the original df. Instead of writing `records_to_update['newcol'] = records_to_update.apply(column_creation_function, axis = 1)`, you can very much write `records_to_update.apply(function_with_side_effects, axis = 1)` as a line on its own that triggers the desired function with the desired side effects for each row in the df.

Reply:
Agree with rahul here. 

Take this with a grain of salt but I found an article comparing a bunch of ways to apply a function over a DF (https://towardsdatascience.com/apply-function-to-pandas-dataframe-rows-76df74165ee4), apparently itertuples and list comp perform better than apply??

Reply:
@atatmaja seems to vary by use case / function called and optimizability. See https://towardsdatascience.com/how-to-make-your-pandas-loop-71-803-times-faster-805030df4f06